frontend/class-frontend.php

```diff
@@ -988,7 +988,7 @@ private function generate_canonical() {
 		 *
 		 * @api string $canonical The canonical URL
 		 */
-		$this->canonical = apply_filters( 'wpseo_canonical', $canonical );
+		$this->canonical = apply_filters( 'wpseo_canonical', user_trailingslashit( $canonical ) );
```

Comment:
It looks like changing the location where `user_trailingslashit` is called will accomplish the same thing and not mess up the Query String. I submitted a change to the PR.